Doc/library/urllib.request.rst

```diff
@@ -1557,9 +1549,38 @@ some point in the future.
    :synopsis: Response classes used by urllib.
 
 The :mod:`urllib.response` module defines functions and classes which define a
-minimal file like interface, including ``read()`` and ``readline()``. The
-typical response object is an addinfourl instance, which defines an ``info()``
-method and that returns headers and a ``geturl()`` method that returns the url.
-Functions defined by this module are used internally by the
-:mod:`urllib.request` module.
+minimal file-like interface, including ``read()`` and ``readline()``.
+Functions defined by this module are used internally by the :mod:`urllib.request` module.
+The typical response object is a :class:`urllib.response.addinfourl` instance:
+
+.. class:: urllib.response.addinfourl
+
+.. attribute:: addinfourl.url
+
+   URL of the resource retrieved, commonly used to determine if a redirect was followed.
+
+.. attribute:: addinfourl.headers
+
+   Returns the meta-information of the page, such as headers,
+   in the form of an :func:`email.message_from_string` instance (see
+   `Quick Reference to HTTP Headers <http://jkorpela.fi/http.html>`_)
+
+.. attribute:: addinfourl.code
+
+   Status code returned by server.
+
+.. attribute:: addinfourl.status
+
+   Status code returned by server. Recommended to use :attr:`~addinfourl.status` instead.
```

Comment:
* The recommendation phrase :attr:`~addinfourl.status` links back to itself while building the docs locally. Also below methods `geturl`, `info` also link to `addinfourl.status` but it's documented to return only status code. Am I missing something here?
* `status` seems to be a newly added property. Does this need a `versionadded` directive?

Reply:
Ha this is recursive!  Was probably copied from the `getstatus` docs.

Reply:
Oops, that was a typo. Added a `versionadded` directive.